Extensions/ParticleSystem/particleemitterobject.ts

```diff
@@ -680,75 +680,75 @@ namespace gdjs {
       }
     }
 
-    getParticleSize2() {
+    getParticleSize2(): float {
       return this.size2;
     }
 
-    setParticleSize2(size): void {
+    setParticleSize2(size: float): void {
       if (this.size2 !== size) {
         this._sizeDirty = true;
         this.size2 = size;
       }
     }
 
-    getParticleAlpha1() {
+    getParticleAlpha1(): number {
       return this.alpha1;
     }
 
-    setParticleAlpha1(alpha): void {
+    setParticleAlpha1(alpha: number): void {
       if (this.alpha1 !== alpha) {
         this._alphaDirty = true;
         this.alpha1 = alpha;
       }
     }
 
-    getParticleAlpha2() {
+    getParticleAlpha2(): number {
       return this.alpha2;
     }
 
-    setParticleAlpha2(alpha): void {
+    setParticleAlpha2(alpha: number): void {
       if (this.alpha2 !== alpha) {
         this._alphaDirty = true;
         this.alpha2 = alpha;
       }
     }
 
-    noMoreParticles() {
+    noMoreParticles(): boolean {
       this._renderer.stop();
```

Comment:
The two commits are for now adding types - which is great! But not fixing the mentioned bugs right?

Reply:
Yes, I forgot to mention it's still a draft. I wanted the issues to mention that we are working on these issues